cdap-common/src/main/java/co/cask/cdap/common/io/Locations.java

```diff
@@ -341,9 +341,9 @@ public static Location getParent(Location location) {
   }
 
   /**
-   * Returns the relative path for a given location
-   * @param locationFactory
-   * @param locationURI
+   * Returns the relative path for a given location URI
+   * @param locationFactory locationFactory to be used to create base URI
+   * @param locationURI URI to be used for base URI
    * @return
    */
   public static String getRelativePath(LocationFactory locationFactory, URI locationURI) {
```

Comment:
didn't catch this earlier, but the URIs that you use to relativize should only contain the path. If the hosts don't match, this will just return the locationURI's path.